### PR TITLE
Transfer Hotspot

### PR DESCRIPTION
**Pull request by tyler-whitman**
Enables transfers of hotspots
- Initiate a transfer from hotspot settings. This send a notification to the buyer
- Buyer can access the transfer from their notifications and submit it
- Seller can cancel transfers
- Checks seller doesn't change the amount while buyer is submitting
- Checks for cancellations on buyers side
- Check for nonce issues on buyers side (buyer does a payment while the transfer is pending)

Currently known issues:
1. Hotspots don't refresh automatically after transfer (@allenan is working on making sure data doesnt get stale)
2. The pending transaction after you submit the transaction isnt working yet
3. Push notification are currently not working, but the notification will show in the list

### CODE REVIEW

**Comment by allenan** [comment]
> * Enables transfers of hotspots
> 
> Currently two known issues:
> 
> 1. Hotspots don't refresh automatically after transfer (@allenan is working on making sure data doesnt get stale)
> 2. The pending transaction after you submit the transaction isnt working yet

yeah hoping to address both of those issues in my branch

File: src/features/hotspots/transfers/TransferRequests.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+export type Transfer = {
+  amountToSeller?: Balance<NetworkTokens>
+  partialTransaction: string
+  gateway: string
+  seller: string
+  buyer: string
+}
+
+export const createTransfer = async (
```
Comment (allenan) [inline src/features/hotspots/transfers/TransferRequests.tsx:24]: not a big deal, but just wondering if this is the right place for this file? would it be more at home in `utils` or even as a redux slice?

File: src/features/hotspots/transfers/HotspotTransfer.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  const hotspotName = animalName(hotspot.address)
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+  const [typedName, setTypedName] = useState('')
+
+  const handleTypeName = (text: string) => {
+    setTypedName(text)
+  }
+
+  const typedHotspotName = () => {
```
Comment (allenan) [inline src/features/hotspots/transfers/HotspotTransfer.tsx:34]: i was confusing this with typescript types in my head for a moment. of course i can't immediately think of a better name off the top of my head 😛

File: src/features/wallet/send/SendView.tsx [hunk 3]
```diff
@@ -142,7 +233,97 @@ const SendView = ({ scanResult }: { scanResult?: QrScanResult }) => {
     triggerNavHaptic()
   }
 
-  const constructTxn = async (): Promise<string> => {
+  const handleSellerTransfer = async (): Promise<undefined> => {
```
Comment (allenan) [inline src/features/wallet/send/SendView.tsx:238]: is `Promise<undefined>` required here? could probably leave it off and it'd be inferred from being an async function

**Review by allenan — APPROVED** [review]
code looks good. we can test the functionality once it's on testflight

File: src/features/hotspots/transfers/TransferRequests.tsx (hunk 1 above)
Comment (tyler-whitman) [reply src/features/hotspots/transfers/TransferRequests.tsx:24]: Hmm yeah good point, I probably should move the data to redux as right now its integrated into the components. I think in regards to time I would leave it as it is for now. Most of the data is already in the SendComponent anyway. 

I kind of like having all the transfer stuff in one folder so its easier to find. I could move this to utils though?

File: src/features/hotspots/transfers/HotspotTransfer.tsx (hunk 2 above)
Comment (tyler-whitman) [reply src/features/hotspots/transfers/HotspotTransfer.tsx:34]: Haha hmm, I could make it `enteredHotspotName`? Its the name they are typing to confirm the transfer

File: src/features/wallet/send/SendView.tsx (hunk 3 above)
Comment (tyler-whitman) [reply src/features/wallet/send/SendView.tsx:238]: Good point, removed it

File: src/features/hotspots/transfers/TransferRequests.tsx (hunk 1 above)
Comment (tyler-whitman) [reply src/features/hotspots/transfers/TransferRequests.tsx:24]: Going to merge it now so people can start testing and we can address this is a separate PR if we want to move it to redux

**Pull request merged**